### PR TITLE
Don't use old testing packages in CI

### LINKED ISSUE #11362: Always use the latest testing packages
## 🚀 Enhancement in CI
Currently, "oldest" testing uses the oldest versions of each testing packages. For example, `pytest==6.0.0` is used in "oldest" setting where its stable version is `pytest==6.2.5` as of now.

You can compare which versions are used in "oldest" and "latest" settings:
- `cpu (ubuntu-18.04, 3.7, oldest, stable) ` https://github.com/PyTorchLightning/pytorch-lightning/runs/4737256599?check_suite_focus=true
- `cpu (ubuntu-18.04, 3.7, latest, stable)` https://github.com/PyTorchLightning/pytorch-lightning/runs/4737256642?check_suite_focus=true

### Motivation
It'll become less likely to spend time debugging something unrelated to our side.

### Pitch
Remove:
https://github.com/PyTorchLightning/pytorch-lightning/blob/59a7ba760548baadf6dbb30864b54cb01c7225a3/.github/set-min-requirements.py#L5

### Alternatives
Use old versions of testing packages, e.g. pytest, (which are likely to have some bugs in testing packages

### Additional context

______________________________________________________________________

#### If you enjoy Lightning, check out our other projects! ⚡

<sub>

- [**Metrics**](https://github.com/PyTorchLightning/metrics): Machine learning metrics for distributed, scalable PyTorch applications.

- [**Lite**](https://pytorch-lightning.readthedocs.io/en/latest/starter/lightning_lite.html): enables pure PyTorch users to scale their existing code on any kind of device while retaining full control over their own loops and optimization logic.

- [**Flash**](https://github.com/PyTorchLightning/lightning-flash): The fastest way to get a Lightning baseline! A collection of tasks for fast prototyping, baselining, fine-tuning, and solving problems with deep learning.

- [**Bolts**](https://github.com/PyTorchLightning/lightning-bolts): Pretrained SOTA Deep Learning models, callbacks, and more for research and production with PyTorch Lightning and PyTorch.

- [**Lightning Transformers**](https://github.com/PyTorchLightning/lightning-transformers): Flexible interface for high-performance research using SOTA Transformers leveraging Pytorch Lightning, Transformers, and Hydra.

</sub>

cc @carmocca @akihironitta @borda

### PR DESCRIPTION
**Pull request by akihironitta**
## What does this PR do?
Fixes #11362.

### Does your PR introduce any breaking changes? If yes, please list them.
None

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [n/a] Did you make sure to **update the documentation** with your changes? (if necessary)
- [n/a] Did you write any **new necessary tests**? (not for typos and docs)
- [n/a] Did you verify new and **existing tests pass** locally with your changes?
- [n/a] Did you list all the **breaking changes** introduced by this pull request?
- [n/a] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
:star_struck: :star_struck: :star_struck: 

### Additional comment
Please add this to an appropriate milestone...

cc @carmocca @akihironitta @borda

### CODE REVIEW

File: .github/set-min-requirements.py [hunk 1]
```diff
@@ -2,8 +2,8 @@
     "requirements.txt",
     "requirements/extra.txt",
     "requirements/loggers.txt",
-    "requirements/test.txt",
     "requirements/examples.txt",
+    # "requirements/test.txt",  # Don't use old testing packages
```
Comment (akihironitta) [inline .github/set-min-requirements.py:6]: Just commenting it out in case someone thinks it's missing, but could also be deleted completely.

**Review by Borda — APPROVED** [review]

**Review by s-rog — APPROVED** [review]
